File: UVa/UVa10763/10763.py
from collections import defaultdict

people = defaultdict(int)
while True:
    n = int(input())
    if n == 0:
        break
    people.clear()
    for i in range(n):
        k = tuple(input().split())
        if people[k[::-1]] > 0:
            people[k[::-1]] -= 1
        else:
            people[k] += 1
    if all(j == 0 for j in people.values()):
        print("YES")
    else:
        print("NO")

File: UVa/UVa10763/10763.cpp
#include <bits/stdc++.h>
using namespace std;
map<pair<int, int>, int> people;
using T = const decltype(people)::value_type &;
int main()
{
    int n;
    while (scanf("%d", &n) && n)
    {
        int from, to;
        people.clear();
        while (n--)
        {
            scanf("%d%d", &from, &to);
            if (people[{to, from}] > 0)
            {
                people[{to, from}]--;
            }
            else
            {
                people[{from, to}]++;
            }
        }
        if (all_of(people.begin(), people.end(), [](T i) { return i.second == 0; }))
        {
            printf("YES\n");
        }
        else
        {
            printf("NO\n");
        }
    }
    return 0;
}
